### PR TITLE
feat: added reject unsinged commits policy

### PR DESCRIPTION
**Pull request by nadav-legit**
#### What's being changed?
Added signed commits verification policy for gitlab

#### Check off the following:

- [X] This PR follows the CONTRIBUTION.md guidelines
- [X] I have self-reviewed my changes before submitting the PR

### CODE REVIEW

File: internal/collectors/gitlab/repository_collector.go [hunk 1]
```diff
@@ -159,6 +159,16 @@ func (rc *repositoryCollector) extendProjectWithWebhooks(project gitlab_collecte
 	return extendedProject
 }
 
+func (rc *repositoryCollector) extendProjectWithPushRul(project gitlab_collected.Repository) gitlab_collected.Repository {
```
Comment (noamd-legit) [inline internal/collectors/gitlab/repository_collector.go:162]: Rul -> Rule

File: internal/collectors/gitlab/repository_collector.go [hunk 2]
```diff
@@ -159,6 +159,16 @@ func (rc *repositoryCollector) extendProjectWithWebhooks(project gitlab_collecte
...
+	rules, _, err := rc.Client.Client().Projects.GetProjectPushRules(project.ID)
+	if err != nil {
+		log.Printf("failed to get project push rule %s", err)
+	}
+	extendedProject := project
+	extendedProject.PushRules = rules
```
Comment (noamd-legit) [inline internal/collectors/gitlab/repository_collector.go:168]: Don't use rules if err != nil. It's not clear what the value will be

File: policies/gitlab/repository.rego [hunk 3]
```diff
@@ -132,3 +132,15 @@ default no_conversation_resolution = false
...
+# METADATA
+# scope: rule
+# title: Unsinged Commits Are Not Allowed
+# description: Require all commits to be signed and verified
+# custom:
+#   remediationSteps: [Make sure you have owner permissions, Go to the projects's settings -> Repository page, Enter "Push Rules" tab. Set the "Reject unsigned commits" checkbox ]
+#   severity: LOW
+default no_signed_commits = false
+no_signed_commits {
+    input.push_rules.reject_unsigned_commits == false
```
Comment (noamd-legit) [inline policies/gitlab/repository.rego:145]: how do you handle the case where input.push_rules == null or input.push_rules.reject_unsigned_commits == null

File: test/repository_test.go [hunk 4]
```diff
@@ -485,3 +485,24 @@ func TestGitlabResolvedThreads(t *testing.T) {
...
+	falseCase := &gitlab2.ProjectPushRules{RejectUnsignedCommits: true}
+	trueCase := &gitlab2.ProjectPushRules{RejectUnsignedCommits: false}
+	options := map[bool]*gitlab2.ProjectPushRules{
+		false: falseCase,
+		true:  trueCase,
+	}
+
+	for _, expectFailure := range bools {
+		flag := options[expectFailure]
+		repositoryTestTemplate(t, name, makeMockData(flag), testedPolicyName, expectFailure, scm_type.GitLab)
```
Comment (noamd-legit) [inline test/repository_test.go:506]: test null cases

**Review by noamd-legit — CHANGES_REQUESTED** [review]

File: policies/gitlab/repository.rego (hunk 3 above)
Comment (nadav-legit) [reply policies/gitlab/repository.rego:145]: 1. good point, adding a check
2. input.push_rules.reject_unsigned_commits will never be null

**Review by noamd-legit — APPROVED** [review]

**Pull request merged**